### PR TITLE
Create SQLAlchemy engine from connection in DB Hook and added autocommit param to insert_rows method

### PR DESCRIPTION
**Pull request by dabla**
Refactored the get_sqlalchemy_engine method in DbAPiHook so it's using the uri as well as the connection for creating an engine, which in most cases nowadays isn't working for a lot of database dialects anyway and which also bypasses the know issues with the get_uri method in combination with sqlalchemy that will be fixed in [#39572](https://github.com/apache/airflow/pull/39572).

Beside that this PR also added the autocommit parameter to the insert_rows method of the DbAPiHook, just like it is already the case with the run method.  This is a preparation for a new PR that will introduce an SQLInsertRowsOperator, which will be a convenient operator allowing you to insert rows using an operator instead of the need to use a PythonOperator in combination with the already existing insert_rows method of the DbApiHook, that way it will be directly possible to pass an XCom containing the rows to persist to the operator which aren't directly feasible by using the GenericTransfer operator, for example in the case of have to download data from a REST endpoint and persisting it directly to a database.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Small request :). PLEASE make sure to have empty line before `--` in issue description - otherwise the last paragraph is rendered as heading in GitHub issue :D

**Comment by dabla** [comment]
The documentation build failed here: https://github.com/apache/airflow/actions/runs/10004956096/job/27654856973?pr=40669

But when I run it locally with breeze no issues at all.

**Comment by potiuk** [comment]
failed when fatching remote inventory.

File: airflow/settings.py [hunk 1]
```diff
@@ -256,6 +256,16 @@ def __init__(self):
     def remove(*args, **kwargs):
         pass
 
+    def get_bind(
```
Comment (potiuk) [inline airflow/settings.py:259]: The failing tests should be marked as @pytest.mark.db_test instead. Not sure which tests failed with it and why (somewhat the exception above would have to be ignored), but This session should have never been initialized in the "non-db" tests cases. See https://github.com/apache/airflow/blob/main/contributing-docs/testing/unit_tests.rst#db-and-non-db-tests for more explanation

Comment (dabla) [reply airflow/settings.py:259]: Will see if they fail again, don't remember which tests started failling...

**Review by potiuk — APPROVED** [review]
I really like the idea implemented here.

**Comment by potiuk** [comment]
Rebased to see if it still works in main.

**Comment by dabla** [comment]
> Rebased to see if it still works in main.

Thank you @potiuk, cannot check it myself right now on vacation 😉

**Review by potiuk — APPROVED** [review]

**Pull request merged**